core/src/ldml/C7043_ldml.md

```diff
@@ -324,10 +324,87 @@ For each key:
 - `vkey`: Is the standard vkey, 0-255
 - `target`: Is the target (resolved) vkey, 0-255.
 
+### C7043.2.13 `layr`—Layers list
+
+Represents layers on the keyboard.
+- Each key entry corresponds to a key on the row
+
+| ∆ | Bits | Name       | Description                              |
+|---|------|------------|------------------------------------------|
+| 0 |  32  | ident      | `layr`                                   |
+| 4 |  32  | size       | int: Length of section                   |
+| 8 |  32  | listCount  | int: Number of layer lists               |
+|12 |  32  | layerCount | int: number of layer entries             |
+|16 |  32  | rowCount   | int: number of row entries               |
+|20 |  32  | keyCount   | int: number of key entries               |
+|24 |  64  | reserved   | padding                                  |
+
+Each layer list corresponds to one `<layers>` element.
+There are `listCount` total lists.
+
+| ∆ | Bits | Name       | Description                                |
+|---|------|------------|--------------------------------------------|
+| 0+|  32  | flags      | int: per-layers options                    |
+| 4+|  32  | hardware   | str: layout (`us`,`iso`,`jis`,`abnt2`)     |
+| 8+|  32  | layer      | int: index to first layer element          |
+|12+|  32  | count      | int: number of layer elements in this list |
+
+- `flags`: a 32-bit bitfield defined as below:
+
+  | Bit position | Meaning  |  Description         |
+  |--------------|----------|----------------------|
+  |       0      | form     | 0: hardware          |
+  |       0      | form     | 1: touch             |
+
+Each layer entry corresponds to one `<layer>` element
+There are `layerCount` total layer entries.
+
+| ∆ | Bits | Name       | Description                              |
+|---|------|------------|------------------------------------------|
+| 0+|  32  | id         | str: layer id such as `base` or `shift`  |
+| 4+|  32  | modifier   | str: modifier string                     |
+| 8+|  32  | row        | int: index into rows area (next section) |
+|12+|  32  | count      | int: number of `rows` elements for this layer  |
+
+Each row entry corresponds to one `<row>` element
+There are `rowCount` total row entries.
+
+| ∆ | Bits | Name       | Description                            |
+|---|------|------------|----------------------------------------|
+| 0+|  32  | key        | int: index into key element            |
+| 4+|  32  | count      | int: count of key elements in this row |
+
+Each key entry corresponds to a key in the row.
+There are `keyCount` total key entries.
+
+| ∆ | Bits | Name    | Description                              |
+|---|------|---------|------------------------------------------|
+| 0+|  32  | vkey    | int: vkey ID                             |
```

Comment:
I'm not sure there is a 1-to-1 between vkey and key bag items? e.g. <kbd>s</kbd> vs <kbd>S</kbd> have the same vkey and differ only by modifier. We probably need a key bag index?

Reply:
that would also future-proof us if there are other distinguishing features of a key, some future flag and so on.

Reply:
```suggestion
| 0+|  32  | key     | int: index into `keys` element           |
```

Reply:
Instead of `` `keys` element``, we want `` `keys` section``? But I don't think it currently exists as we want it -- the current `keys` section is designed around vkey+modifier and is optimized for hardware layout.

Reply:
OK. May need a new keys section then (`key2`) that is designed around the entire data set- flicks, longpress and so on. This is where the duplication comes in which we expected.

Reply:
I wonder if renaming the original `keys` section to `hkey` (aka "hardware key") is worth the hassle?

Reply:
The difference is still going to need to be explained. So I'm not sure it really buys much. In any event, I would rename after key2 is already in place.

It also might be that if key2 has vkey information, we can eventually drop the original keys array. Not a primary goal.